### PR TITLE
Simplify exception handling in block worker

### PR DESCRIPTION
**Pull request by rongrong**
This change removes a lot of unnecessary Exceptions in block worker and underlying block store APIs.

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java
Comment (jja725) [inline core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java:507]: I think it's a good idea to aggregate those exceptions but might be more insightful to provide which one goes wrong when debugging

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java
Comment (jja725) [inline core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java:369]: update interface so the caller above aware of this change

**Review by jja725 — COMMENTED** [review]
Thanks for this great improvement! I think there's some specific handling in the upper level we need to pay attention. For example, some code directly catches BlockDoesNotExistException and ignore it. So not in all situation we throw runtime exception to the user in original implementation(although we might want to surface that exception).

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java
Comment (jja725) [inline core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java:75]: IllegalStateException seems safer

File: core/server/worker/src/main/java/alluxio/worker/block/CacheRequestManager.java
Comment (jja725) [inline core/server/worker/src/main/java/alluxio/worker/block/CacheRequestManager.java:292]: catch all IllegalState doesn't seems good here

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java
Comment (rongrong) [reply core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java:497]: It is actually checking whether committed block exists or not. So the name is accurate.

Comment (jja725) [reply core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java:497]: I'm suggesting here since I'm a little confused about why we need to checkBlockDoes not exist when we abort the block until I look at the code.

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java
Comment (beinan) [inline core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java:502]: How about inline these method?  your call

**Review by beinan — APPROVED** [review]
looks good to me, just one non-blocking comment

**Review by jja725 — APPROVED** [review]
leave minor comments, good to go after fix checkstyle.

**Comment by rongrong** [comment]
alluxio-bot, merge this please

**Pull request merged**
